Observer/SalesQuoteProductAddAfter.php

```diff
@@ -0,0 +1,157 @@
+
+    /**
+     * Magento Category Factory
+     * @var CategoryFactory $_categoryFactory
+     */
+    protected $_categoryFactory;
+
+    /**
+     * Magento Checkout Session data
+     * @var Session $_checkoutSession
+     */
+    protected $_checkoutSession;
+
+    /**
+     * SalesQuoteProductAddAfter constructor.
+     * @param CartSearchRepository $cartSearchRepository
+     * @param CategoryFactory $categoryFactory
+     * @param Session $checkoutsession
+     */
+    public function __construct(
+        CartSearchRepository $cartSearchRepository,
+        CategoryFactory $categoryFactory,
+        Session $checkoutsession
+    )
+    {
+        $this->_cartSearchRepository = $cartSearchRepository;
+        $this->_categoryFactory = $categoryFactory;
+        $this->_checkoutSession = $checkoutsession;
+    }
+
+    public function execute(Observer $observer)
+    {
+        $quote = $observer->getData('items')[0]->getQuote();
+        $addedItems = $observer->getData('items');
+
+        foreach ( $addedItems as $item ){
+            $this->klAddedToCartItemData( $quote, $item );
+        }
+    }
+
+    /**
+     * Build Added to Cart payload for current Added item
+     * @param $quote
+     * @param $addedItem
+     */
+    public function klAddedToCartItemData( $quote, $addedItem )
+    {
+        $addedProduct = $addedItem->getProduct();
+        $addedItemData = array(
+            'AddedItemCategories' => (array) $this->getCategoryName( $addedProduct->getCategoryIds() ),
+            'AddedItemDescription' => (string) strip_tags( $addedProduct->getDescription() ),
+            'AddedItemImageUrlKey' => (string) $addedProduct->getData('small_image'),
+            'AddedItemPrice' => (float) $addedProduct->get_price(),
+            'AddedItemQuantity' => (int) $addedItem->getQty(),
+            'AddedItemProductID' => (int) $addedProduct->getId(),
+            'AddedItemProductName' => (string) $addedItem->getName(),
+            'AddedItemSku' => (string) $addedProduct->getSku(),
+            'AddedItemUrl' => (string) $addedProduct->getProductUrl()
+        );
+
+        $klAddedToCartPayload = array_merge( $this->klBuildCartData( $quote, $addedItem ), $addedItemData );
+
+        $this->getCheckoutSession()->setKlAddedToCartKey( $klAddedToCartPayload );
+    }
+
+    /**
+     * Helper function to add items from cart to Added to Cart payload
+     * @param $quote
+     * @param $addedItem
+     * @return array
+     */
+    public function klBuildCartData( $quote, $addedItem )
+    {
+        $cartItems = $quote->getItems() ? $quote->getitems() : array();
+        $cartQty = 0;
+        $items = array();
+        $cartItemNames = array();
+        $cartItemCategories = array();
+
+        foreach( $cartItems as $item ) {
+            $product = $item->getProduct();
+            $cartItemId = $product->getId();
+            $itemCategories = $this->getCategoryName( $product->getCategoryIds() );
```

Comment:
Nit: the additional spacing is a Wordpress specific style convention. We don't need those here
```suggestion
            $itemCategories = $this->getCategoryName($product->getCategoryIds());
```